jscomp/snapshot.ninja

```diff
@@ -37,7 +37,9 @@ o $SNAP/unstable/js_compiler.ml: bspack | ./bin/bspack.exe $LTO
     flags = -D BROWSER=true  -MD    -I ml  $includes
     main  = Jsoo_main
 
-
+o $SNAP/unstable/js_playground_compiler.ml: bspack | ./bin/bspack.exe $LTO
```

Comment:
Not sure if there's a way to do conditional rules to add the playground snapshot build

Reply:
@cristianoc compilation performance should be generally good, so running this with every ninja build feels like the right thing to do. Also the `jsoo_main` entrypoint is identical from a dependency perspective and always has been part of the build process ever since.

Note: The slowest part of building the playground is running jsoo with `scripts/repl.js`, and we'd probably want to set up a CI job later to automate / test the bundle build process.

Reply:
here are some benchmarks:

**BASELINE: no snapshot rule for any playground**
```
scripts/ninja.js config && time scripts/ninja.js build
config for the first time may take a while
rescript: no work to do.
rescript: [983/983] test/reactDOMRe.cmi
rescript: [2/2] ../lib/4.06.1/whole_compiler.ml
POST-PROCESSING-FILE: ../lib/4.06.1/whole_compiler.ml
config for the first time may take a while
rescript: no work to do.
scripts/ninja.js build  10.22s user 4.53s system 291% cpu 5.057 total
```

**TEST 2: snapshot rule for jsoo_main only**
```
scripts/ninja.js config && time scripts/ninja.js build
config for the first time may take a while
rescript: no work to do.
rescript: [983/983] test/unsafe_this.cmj
rescript: [3/3] ../lib/4.06.1/whole_compiler.ml
POST-PROCESSING-FILE: ../lib/4.06.1/whole_compiler.ml
config for the first time may take a while
rescript: no work to do.
scripts/ninja.js build  11.26s user 4.75s system 310% cpu 5.147 total
```

**TEST 1: snapshot rule for jsoo_main and jsoo_playground_main**
```
scripts/ninja.js config && time scripts/ninja.js build
config for the first time may take a while
rescript: no work to do.
rescript: [983/983] test/flow_parser_reg_test.cmj
rescript: [4/4] ../lib/4.06.1/whole_compiler.ml
POST-PROCESSING-FILE: ../lib/4.06.1/whole_compiler.ml
config for the first time may take a while
rescript: no work to do.
scripts/ninja.js build  12.24s user 4.74s system 268% cpu 6.324 total
```

Reply:
Looks like the differences are absorbed by measurement noise, so nothing of note there.